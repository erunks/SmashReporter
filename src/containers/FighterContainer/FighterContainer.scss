@import '../../styles/base.scss';

.FighterContainer {
  @include absolute-position;

  align-items: center;
  display: grid;
  grid-template-columns: 10% auto 10%;
  justify-content: center;
  // flex-direction: row;
}
