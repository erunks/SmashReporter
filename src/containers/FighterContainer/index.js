import React from 'react';
import { useRecoilValue, useSetRecoilState } from 'recoil';
import { selectedFightersState } from '../../recoils/fighter';
import { selectingPlayerState } from '../../recoils/player';
import FighterChoice from '../../components/FighterChoice';
import FighterSelection from '../../components/FighterSelection';
import './FighterContainer.scss';

const FighterContainer = () => {
  const selectedFighters = useRecoilValue(selectedFightersState);
  const setSelectingPlayer = useSetRecoilState(selectingPlayerState);

  return (
    <div className="FighterContainer">
      <FighterChoice
        leftFacing={true}
        fighter={selectedFighters[0]}
        onClick={() => setSelectingPlayer(0)}
      />
      <FighterSelection/>
      <FighterChoice
        fighter={selectedFighters[1]}
        onClick={() => setSelectingPlayer(1)}
      />
    </div>
  )
};

export default FighterContainer;
