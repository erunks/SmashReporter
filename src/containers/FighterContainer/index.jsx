import React, { useState } from 'react';
import FighterChoice from '../../components/FighterChoice/index.jsx';
import FighterSelection from '../../components/FighterSelection/index.jsx';
import './FighterContainer.scss';

const FighterContainer = () => {
  const [playerOneFighter, setplayerOneFighter] = useState('mario');
  const [playerTwoFighter, setplayerTwoFighter] = useState('mario');
  const [selectingPlayer, setSelectingPlayer] = useState(0);


  const setPlayerFighter = (name) => {
    if (selectingPlayer === 0) {
      setplayerOneFighter(name);
    } else if (selectingPlayer === 1) {
      setplayerTwoFighter(name);
    }
  };

  return (
    <div className="FighterContainer">
      <FighterChoice
        leftFacing={true}
        fighter={playerOneFighter}
        onClick={() => setSelectingPlayer(0)}
      />
      <FighterSelection
        setPlayerFighter={setPlayerFighter}
        selectingPlayer={selectingPlayer}
      />
      <FighterChoice
        fighter={playerTwoFighter}
        onClick={() => setSelectingPlayer(1)}
      />
    </div>
  )
};

export default FighterContainer;
