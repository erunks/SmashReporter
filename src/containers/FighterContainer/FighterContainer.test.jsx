import React from 'react';
import { shallow } from 'enzyme';
import FighterContainer from './index.jsx';

describe('<FighterContainer>', () => {
  const container = shallow(<FighterContainer />);

  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });
});
