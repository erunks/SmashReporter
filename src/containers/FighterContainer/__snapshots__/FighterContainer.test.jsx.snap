// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FighterContainer> should match the snapshot 1`] = `"<div class=\\"FighterContainer\\"><div class=\\"FighterChoice faceLeft\\" role=\\"button\\"><img src=\\"https://www.smashbros.com/assets_v2/img/fighter/thumb_v/mario.png\\" alt=\\"fighter: mario\\"/><div class=\\"helpText\\"><p>Click to Select Fighter</p></div></div><div class=\\"FighterSelection\\">0</div><div class=\\"FighterChoice \\" role=\\"button\\"><img src=\\"https://www.smashbros.com/assets_v2/img/fighter/thumb_v/mario.png\\" alt=\\"fighter: mario\\"/><div class=\\"helpText\\"><p>Click to Select Fighter</p></div></div></div>"`;
